rama-http/src/headers/x_robots_tag/element.rs

```diff
@@ -0,0 +1,46 @@
+use crate::headers::x_robots_tag::rule::Rule;
+use rama_core::error::OpaqueError;
+use std::fmt::Formatter;
+use std::str::FromStr;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Element {
+    bot_name: Option<String>, // or `rama_ua::UserAgent` ???
+    indexing_rule: Rule,
+}
+
+impl std::fmt::Display for Element {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match &self.bot_name {
+            None => write!(f, "{}", self.indexing_rule),
+            Some(bot) => write!(f, "{}: {}", bot, self.indexing_rule),
+        }
+    }
+}
+
+impl FromStr for Element {
+    type Err = OpaqueError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let (bot_name, indexing_rule) = match Rule::from_str(s) {
+            Ok(rule) => (None, Ok(rule)),
+            Err(e) => match *s.split(":").map(str::trim).collect::<Vec<_>>().as_slice() {
```

Comment:
You'll need to anyway modify the Element code, so this might dissapear, but as extra info, be careful with collecting stuff in a Vec, just for tmp reasons. Gives allocations for not much good. Instead better to split fixed into a tuple, or use the iterator directly .